src/runtime/profiling.cc

```diff
@@ -234,135 +391,59 @@ std::string FormatTable(const std::vector<std::unordered_map<std::string, Object
   return s.str();
 }
 
-String Profiler::Report(bool aggregate, bool sort) {
+std::string DeviceString(Device dev) {
+  return DeviceName(dev.device_type) + std::to_string(dev.device_id);
+}
+
+Report Profiler::Report(bool aggregate, bool sort) {
   std::vector<std::pair<Device, double>> global_times;
   for (auto p : global_timers_) {
     global_times.emplace_back(p.first, p.second->SyncAndGetElapsedNanos() / 1e3);
   }
-  double overall_time = 0.;
+
+  std::unordered_map<String, Map<String, ObjectRef>> overall;
   for (auto p : global_times) {
-    overall_time = std::max(overall_time, p.second);
+    std::unordered_map<String, ObjectRef> row;
+    row["Name"] = String("Total");
+    row["Duration (us)"] = ObjectRef(make_object<DurationNode>(p.second));
+    row["Percent"] = ObjectRef(make_object<PercentNode>(100));
```

Comment:
Should `Pervent` be the total time on one device over all devices? If so, this should not always be 100, right?

Reply:
I fixed the calculation to now be a percent of the longest runtime.